version: "3.9"

services:
    django:
        build: .
        container_name: django
        command:  >
             sh -c "
             python manage.py makemigrations &&
             python manage.py migrate &&
             python manage.py runserver 0.0.0.0:8000"
        volumes:
            - .:/usr/src/app/
        ports:
            - "8000:8000"
        environment:
            - DEBUG=1
            - CELERY_BROKER=redis://redis:6379/0
            - CELERY_BACKEND=redis://redis:6379/0
            - REDIS_URL=redis://redis:6379/0
        depends_on:
            postgres:
                condition: service_healthy
            redis:
                condition: service_started
    postgres:
        image: postgres
        container_name: postgres
        environment:
            - POSTGRES_DB=myproject
            - POSTGRES_USER=user
            - POSTGRES_PASSWORD=password
        ports:
            - "5432:5432"
        healthcheck:
            test: [ "CMD-SHELL", "pg_isready" ]
            interval: 10s
            timeout: 5s
            retries: 5
        volumes:
            - pgdata:/var/lib/postgresql/data/
    redis:
        image: "redis:alpine"
        ports:
            - "6379:6379"
    
    celery:
        build: .
        command: celery -A myproject worker -l INFO
        volumes:
            - .:/usr/src/app
        environment:
            - DEBUG=1
            - CELERY_BROKER=redis://redis:6379/0
            - CELERY_BACKEND=redis://redis:6379/0
        depends_on:
            - django
            - redis
  
volumes:
  pgdata: